[docs] Start documentation for the data grid filter features (#754)
mui-org/material-ui-x: docs/src/pages/components/data-grid/filtering/filtering.md

{{"demo": "pages/components/data-grid/filtering/BasicToolbarFilteringGrid.js", "bg": "inline"}}

### Disable filter

#### Globally

Filters are enabled by default, but you can easily disable this feature by setting the `disableColumnFilter` prop.

```jsx
<DataGrid disableColumnFilter />
```

#### Per column

You can disable the filter on a column by setting the `filterable` property of the `ColDef` to `false`;

```js
const columns = [{ field: 'image', filterable: false }];
```

{{"demo": "pages/components/data-grid/filtering/DisableFilteringGrid.js", "bg": "inline"}}

### Custom filter operator

The data grid supports different operators for the native column types.
However, you can extend the operator and add your own, customize the input component or set your own operator for a new column type.

In this demo, you will see how to reuse the numeric filter and customize the input filter value component.

{{"demo": "pages/components/data-grid/filtering/CustomRatingFilterOperator.js", "bg": "inline"}}

<!-- Demo filter with new column type

Demo filter with new operator -->

<!-- ### Server-side filter -->

<!-- ### Controlled filtering -->

> ### Multi-filtering ⚡️

Reviewer: ```suggestion
### Multi-column filtering ⚡️
```

Reviewer 2: It can be two filters on the same column, does the proposed title covers it?

Reviewer: Hmm... not directly, but "multi-filtering" doesn't read well. Perhaps we mention multiple filters on the same column on the description?

> `XGrid` allows filtering by multiple columns.

Reviewer: ```suggestion
`XGrid` supports filtering by multiple columns.
```

<p class="description">Filtering helps view particular or related records in the Data Grid.</p>

## 🚧 Column filter
## Basic Filtering

> ⚠️ This feature isn't implemented yet. It's coming.
>
> 👍 Upvote [issue #201](https://github.com/mui-org/material-ui-x/issues/201) if you want to see it land faster.
Column filters can be set using the column menu and clicking the filter menu item.
Alternatively, if the grid has the toolbar displayed, just need to click on the filter button.

Column filters are filters that are applied to the data at the column level.
A filtered column can be can pre-configured using the filterModel prop:

> ## 🚧 Quick filter

Author: Yes but we can add it later...

Just wanted to cleanup the page, but I can put it back, not an issue :)

> ## 🚧 Quick filter

Reviewer 2: From my perspective, it depends on the following: do we want to implement this feature no matter what, and do we think it's important?

If the answer is yes, then I think it makes sense to keep linking the issue so we can use it to prioritize the roadmap. If the answer is no, then we probably want to remove the link, to no skewed the 👍 on the lined issue more.

> ## 🚧 Quick filter

Author: We could also add a new page with all the issues in progress so ppl can upvote issues from a central location

> ## 🚧 Quick filter

Reviewer 2: > a new page with all the issues in progress

I think is option 2. We already have this page, it's GitHub's issues, that we could maybe link a bit better, by mentioning it in the roadmap page: https://material-ui.com/discover-more/roadmap/.

{{"demo": "pages/components/data-grid/filtering/BasicFilteringGrid.js", "bg": "inline"}}

### Disable filter

#### Globally

Filters are enabled by default, but you can easily disable this feature by setting the `disableColumnFilter` prop.

> ```ts

Reviewer 2: This snippet relies on JSX
````suggestion
```jsx
````

<DataGrid disableColumnFilter />
```

#### Per column

You can disable the filter on a column by setting the `filterable` prop of the `ColDef` to `false`;

> ```ts

Reviewer 2: There is no TypeScript

````suggestion
```js
````

const columns = [{ field: 'image', filterable: false }];
```

{{"demo": "pages/components/data-grid/filtering/DisableFilteringGrid.js", "bg": "inline"}}

### Custom filter operator

> We currently support different operators for the native column types.

Reviewer 2: ```suggestion
The data grid supports different operators for the native column types.
```
